test/system/030-run.bats

```diff
@@ -1017,8 +1017,13 @@ EOF
 }
 
 @test "podman run --net=host --cgroupns=host with read only cgroupfs" {
-    # verify that the last /sys/fs/cgroup mount is read-only
-    run_podman run --net=host --cgroupns=host --rm $IMAGE sh -c "grep ' / /sys/fs/cgroup ' /proc/self/mountinfo | tail -n 1 | grep '/sys/fs/cgroup ro'"
+    if is_cgroupsv1; then
+        # verify that the memory controller is mounted read-only
+        run_podman run --net=host --cgroupns=host --rm $IMAGE sh -c "grep '/sys/fs/cgroup/memory ro.* cgroup cgroup' /proc/self/mountinfo"
```

Comment:
Works rootful on RHEL8.8, fails rootless.

Could I encourage you to avoid tests like "grep"? They're antihelpful when things go wrong: they give no useful information on error, just "sorry". A different way to write that (and line 1025) could be:
```bash
    run_podman ... grep /sys/fs/cgroup /proc/self/mountinfo       <--- note that this works for both
    if is_cgroupsv1; then
        assert "$output" =~ "something /memory ro"
    else
        assert "$output" =~ "something else ro"
```

In this case, to save you the trouble, here's the root and rootless output of `grep /memory`:
```
# bin/podman run --net=host --cgroupns=host --rm quay.io/libpod/testimage:20221018 grep memory /proc/self/mountinfo
472 461 0:38 /machine.slice/libpod-32a97bee0daf1c4911ea12bd722e9e55d96345d04f39a667fc5edb6b45886f8e.scope /sys/fs/cgroup/memory ro,nosuid,nodev,noexec,relatime - cgroup cgroup rw,seclabel,memory

$ bin/podman run --net=host --cgroupns=host --rm quay.io/libpod/testimage:20221018 grep memory /proc/self/mountinfo
447 428 0:38 / /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,seclabel,memory
474 463 0:38 / /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,seclabel,memory
```

Reply:
yes that looks better, I'll fix it